kafka-connector/src/test/java/com/google/pubsub/kafka/sink/CloudPubSubSinkTaskTest.java

```diff
@@ -362,18 +364,66 @@ public void testFlushWithNoPublishInPut() throws Exception {
    * Tests that if a Future that is being processed in flush() failed with an exception, that an
    * exception is thrown.
    */
-  @Test(expected = RuntimeException.class)
+  @Test
   public void testFlushExceptionCase() throws Exception {
     task.start(props);
     Map<TopicPartition, OffsetAndMetadata> partitionOffsets = new HashMap<>();
     partitionOffsets.put(new TopicPartition(KAFKA_TOPIC, 0), null);
-    List<SinkRecord> records = getSampleRecords();
+    List<SinkRecord> records = getSampleRecords(0);
     ApiFuture<String> badFuture = getFailedPublishFuture();
     when(publisher.publish(any(PubsubMessage.class))).thenReturn(badFuture);
     task.put(records);
+    assertThrows(RuntimeException.class, () -> task.flush(partitionOffsets));
```

Comment:
Use assertThrows instead of expected so the following verifies are actually reached.